### ISSUE TITLE
Haxelib creates ~ folder on Ubuntu

### ISSUE THREAD

**Issue by PeyTy** [HaxeFoundation/haxelib#351]
Tested on 3.2.1

``` php
peyty@peyty-U16041:~$ haxelib setup
Please enter haxelib repository path with write access
Hit enter for default (/usr/share/haxelib)
Path : ~/.haxelib           
haxelib repository is now /home/peyty/~/.haxelib/
```

It _really_ creates `/home/peyty/~/.haxelib/`:

![image](https://cloud.githubusercontent.com/assets/3642643/18222487/598da780-71a1-11e6-9365-2ce434956f3d.png)

---

Expected behavior: ~ is same as /home/peyty, so I expect /home/peyty/.name folder to be created.
From @ibilon 

> you can't do ~/.haxelib/ (even if ~ was correctly detected) because haxelib use a ~/.haxelib file to store data

Okay then, at least ~/.haxelib should show error or something, and ~/.haxe path must work I think.

``` php
peyty@peyty-U16041:~$ ls ~
~                 mm.cfg        Документы    Музыка         Шаблоны
Dropbox           stencylworks  Загрузки     Общедоступные
examples.desktop  Видео         Изображения  Рабочий стол
peyty@peyty-U16041:~$ ls /home/peyty/
~                 mm.cfg        Документы    Музыка         Шаблоны
Dropbox           stencylworks  Загрузки     Общедоступные
examples.desktop  Видео         Изображения  Рабочий стол
```

**Comment by L4D15** [comment]
This is happening to me too in macOS.

**Comment by mlms13** [comment]
If your instincts tell you to `rm ~`, don't do it. I did it and it wasn't pretty. :P

**Comment by paulorobertocruz** [comment]
Is this still a problem? I will try working on this.

**Comment by tobil4sk** [comment]
This was resolved in #382
